speedify.py

```diff
@@ -424,11 +594,348 @@ def safebrowsing_stats():
     return _run_speedify_cmd(args)
 
 
-### Setter functions ###
+#
+# Setter functions
+#
+
+
+@exception_wrapper("Failed to set directory server")
+def directory(domain: str = ""):
+    """
+    directory(domain)
+
+    Uses the given domain as the directory server.
+
+    :param domain: The domain of the directory server.
+    :type operation: str
+    """
+    return _run_speedify_cmd(["directory", domain])
+
+
+@exception_wrapper("Failed to set DNS")
+def dns(ip_addr: str = ""):
+    """
+    dns(ip_addr)
+
+    Uses the given IP address for as the DNS server.
+
+    Example:
+        dns("8.8.8.8")
+
+    :param ip_addr: The IP address of the DNS server.
+    :type operation: str
+    """
+    return _run_speedify_cmd(["dns", ip_addr])
+
+
+@exception_wrapper("Failed to set streaming bypass")
+def streaming_domains(operation: str, domains: str):
+    """
+    streaming_domains(operation, domains)
+
+    Add, remove, or set the streaming hint for some domains.
+
+    Example:
+        streaming_domains("add", "example.com google.com")
+        streaming_domains("rem", "example.com google.com")
+
+    :param operation: The operation to perform. One of:
+        "add"
+        "rem"
+        "set"
```

Comment:
I think instead of passing an operation as a string, it would be more usable to have as separate functions, such as `add_streaming_domain`, `remove_streaming_domain`, `set_streaming_domains`.

This comment applies to a bunch of the other functions below too that follow a similar pattern.

Reply:
I would find it much more usable (documentable, portable, etc) to have the python library as on-par with the CLI as possible.

For the future -- I would find that syntax (and others like it in other places) difficult to automate in the future, should we decide to make this python program automatically generate from the CLI `help`.

Up to you.

Reply:
IOW I think that consistency within the whole of the ecosystem is worthwhile. Either way, let me know.

Reply:
I'll go with the prefix/affix